[CI] CustomAuthorizationEngineTests.java:56: error: no suitable constructor found for Authentication(User,RealmRef,RealmRef)

### CI Link

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+example-plugins/930/console

### Repro line

cd plugin/examples; ./gradlew build --include-build ../..

### Does it reproduce?

Yes

### Applicable branches

master

### Failure history

_No response_

### Failure excerpt

The example projects fails with a compile error caused by this change here: https://github.com/elastic/elasticsearch/pull/86206

This is blocking current PR checks as example pr build is failing.

the stack trace:
> Task :security-authorization-engine:compileTestJava FAILED
09:14:22 /dev/shm/elastic+elasticsearch+pull-request+example-plugins/plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:56: error: no suitable constructor found for Authentication(User,RealmRef,RealmRef)
09:14:22                 new Authentication(new User("joe", new String[]{"custom_superuser"}, new User("bar", "not_superuser")),
09:14:22                 ^
09:14:22     constructor Authentication.Authentication(User,RealmRef,RealmRef,Version,AuthenticationType,Map<String,Object>) is not applicable
09:14:22       (actual and formal argument lists differ in length)
09:14:22     constructor Authentication.Authentication(StreamInput) is not applicable
09:14:22       (actual and formal argument lists differ in length)
09:14:22 /dev/shm/elastic+elasticsearch+pull-request+example-plugins/plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:73: error: no suitable constructor found for Authentication(User,RealmRef,RealmRef)
09:14:22                 new Authentication(new User("joe", new String[]{"not_superuser"}, new User("bar", "custom_superuser")),
09:14:22

--- a/plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java
+++ b/plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java
@@ -52,9 +52,9 @@ public class CustomAuthorizationEngineTests extends ESTestCase {
         CustomAuthorizationEngine engine = new CustomAuthorizationEngine();
         // unauthorized
         {
-            Authentication authentication =
-                new Authentication(new User("joe", new String[]{"custom_superuser"}, new User("bar", "not_superuser")),
-                    new RealmRef("test", "test", "node"), new RealmRef("test", "test", "node"));
+            Authentication authentication = Authentication
+                .newRealmAuthentication(new User("bar", "not_superuser"), new RealmRef("test", "test", "node"))
+                .runAs(new User("joe", "custom_superuser"), new RealmRef("test", "test", "node"));
             RequestInfo info = new RequestInfo(authentication, request, action, null);
             PlainActionFuture<AuthorizationInfo> future = new PlainActionFuture<>();
             engine.resolveAuthorizationInfo(info, future);
@@ -69,9 +69,9 @@ public class CustomAuthorizationEngineTests extends ESTestCase {
 
         // authorized
         {
-            Authentication authentication =
-                new Authentication(new User("joe", new String[]{"not_superuser"}, new User("bar", "custom_superuser")),
-                    new RealmRef("test", "test", "node"), new RealmRef("test", "test", "node"));
+            Authentication authentication = Authentication
+                .newRealmAuthentication(new User("bar", "custom_superuser"), new RealmRef("test", "test", "node"))
+                .runAs(new User("joe", "not_superuser"), new RealmRef("test", "test", "node"));
             RequestInfo info = new RequestInfo(authentication, request, action, null);
             PlainActionFuture<AuthorizationInfo> future = new PlainActionFuture<>();
             engine.resolveAuthorizationInfo(info, future);
@@ -103,7 +103,8 @@ public class CustomAuthorizationEngineTests extends ESTestCase {
         // unauthorized
         {
             RequestInfo unauthReqInfo =
-                new RequestInfo(new Authentication(new User("joe", "not_superuser"), new RealmRef("test", "test", "node"), null),
+                new RequestInfo(
+                    Authentication.newRealmAuthentication(new User("joe", "not_superuser"), new RealmRef("test", "test", "node")),
                     requestInfo.getRequest(), requestInfo.getAction(), null);
             PlainActionFuture<AuthorizationInfo> future = new PlainActionFuture<>();
             engine.resolveAuthorizationInfo(unauthReqInfo, future);
@@ -128,7 +129,8 @@ public class CustomAuthorizationEngineTests extends ESTestCase {
         // authorized
         {
             RequestInfo requestInfo =
-                new RequestInfo(new Authentication(new User("joe", "custom_superuser"), new RealmRef("test", "test", "node"), null),
+                new RequestInfo(
+                    Authentication.newRealmAuthentication(new User("joe", "custom_superuser"), new RealmRef("test", "test", "node")),
                     new SearchRequest(), "indices:data/read/search", null);
             PlainActionFuture<AuthorizationInfo> future = new PlainActionFuture<>();
             engine.resolveAuthorizationInfo(requestInfo, future);
@@ -149,7 +151,8 @@ public class CustomAuthorizationEngineTests extends ESTestCase {
         // unauthorized
         {
             RequestInfo requestInfo =
-                new RequestInfo(new Authentication(new User("joe", "not_superuser"), new RealmRef("test", "test", "node"), null),
+                new RequestInfo(
+                    Authentication.newRealmAuthentication(new User("joe", "not_superuser"), new RealmRef("test", "test", "node")),
                     new SearchRequest(), "indices:data/read/search", null);
             PlainActionFuture<AuthorizationInfo> future = new PlainActionFuture<>();
             engine.resolveAuthorizationInfo(requestInfo, future);
@@ -171,7 +174,7 @@ public class CustomAuthorizationEngineTests extends ESTestCase {
         final String action = "cluster:monitor/foo";
         final TransportRequest request = new TransportRequest() {};
         final Authentication authentication =
-            new Authentication(new User("joe", "custom_superuser"), new RealmRef("test", "test", "node"), null);
+            Authentication.newRealmAuthentication(new User("joe", "custom_superuser"), new RealmRef("test", "test", "node"));
         return new RequestInfo(authentication, request, action, null);
     }
 }

--- a/plugins/examples/settings.gradle
+++ b/plugins/examples/settings.gradle
@@ -11,11 +11,7 @@ plugins {
 }
 
 // Include all subdirectories as example projects
-rootDir.listFiles().findAll { it.directory && new File(it, 'build.gradle').exists() }
-        // filter out failing project till https://github.com/elastic/elasticsearch/issues/86378
-        // is addressed
-        .findAll {projectDir -> (projectDir.name == 'security-authorization-engine') == false }
-        .each { subDir ->
+rootDir.listFiles().findAll { it.directory && new File(it, 'build.gradle').exists() }.each { subDir ->
   include ":${subDir.name}"
 }
 

